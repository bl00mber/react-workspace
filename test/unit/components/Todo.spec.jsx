import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Todo from '@/components/Todo';

describe('Todo', () => {
  test('should match enzyme shallow snapshot', () => {
    const props = {
      completed: false,
      text: 'test 1',
      onClick: jest.fn(),
    };
    const wrapper = shallow(<Todo {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({
      completed: true,
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should call onClick clicking on element', () => {
    const props = {
      completed: false,
      text: 'test 1',
      onClick: jest.fn(),
    };
    const wrapper = shallow(<Todo {...props} />);
    const clickEvent = {};
    wrapper.find('li').simulate('click', clickEvent);
    expect(props.onClick).toHaveBeenCalledWith(clickEvent);
  });
});
