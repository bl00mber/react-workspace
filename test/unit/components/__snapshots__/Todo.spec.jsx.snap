// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match enzyme shallow snapshot 1`] = `
<li
  onClick={[MockFunction]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  test 1
</li>
`;

exports[`Todo should match enzyme shallow snapshot 2`] = `
<li
  onClick={[MockFunction]}
  style={
    Object {
      "textDecoration": "line-through",
    }
  }
>
  test 1
</li>
`;
